pkg/util/api.go

```diff
@@ -26,3 +33,32 @@ func GetAddresses(endpoint *v1.Endpoints) []string {
 	}
 	return serverAddresses
 }
+
+// WaitForAPIServerReady waits for the API Server's /readyz endpoint to report "ok" with timeout.
+// This is cribbed from the Kubernetes controller-manager app, but checks the readyz endpoint instead of the deprecated healthz endpoint.
+func WaitForAPIServerReady(client clientset.Interface, timeout time.Duration) error {
+	var lastErr error
+
+	err := wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		healthStatus := 0
+		result := client.Discovery().RESTClient().Get().AbsPath("/readyz").Do(context.TODO()).StatusCode(&healthStatus)
+		if result.Error() != nil {
+			lastErr = fmt.Errorf("failed to get apiserver /readyz status: %v", result.Error())
+			return false, nil
+		}
+		if healthStatus != http.StatusOK {
+			content, _ := result.Raw()
+			lastErr = fmt.Errorf("APIServer isn't ready: %v", string(content))
+			logrus.Warnf("APIServer isn't ready yet: %v. Waiting a little while.", string(content))
+			return false, nil
+		}
+
+		return true, nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("%v: %v", err, lastErr)
```

Comment:
`errors.Wrap()` ?

Reply:
I copy-pasted this from upstream and just `s/health/ready/` but yeah there are probably a couple enhancements that I could make to it.

Reply:
That said, I'm not sure that wrapping works that way. This would more properly be a multierr use case I think?